app/models/roster.rb

```diff
@@ -10,10 +10,14 @@ class Roster < ApplicationRecord
 
   has_many :assignments, dependent: :destroy
   has_many :memberships, dependent: :destroy
+
+  # Covered by the dependent: :destroy on memberships
+  # rubocop:disable Rails/HasManyOrHasOneDependent
   has_many :admin_memberships, -> { where(admin: true) },
            class_name: 'Membership', inverse_of: :roster
   has_many :non_admin_memberships, -> { where.not(admin: true) },
            class_name: 'Membership', inverse_of: :roster
+  # rubocop:enable Rails/HasManyOrHasOneDependent
```

Comment:
Rubocop will also accept a `dependent: nil` option here as a confirmation of "yes I want to do nothing" for this cop (it's what we do on the hub for the flex models).

Maybe just a little more sightly than disabling the linter altogether?

```suggestion
  has_many :admin_memberships, -> { where(admin: true) },
           class_name: 'Membership', dependent: nil, inverse_of: :roster
  has_many :non_admin_memberships, -> { where.not(admin: true) },
           class_name: 'Membership', dependent: nil, inverse_of: :roster
```